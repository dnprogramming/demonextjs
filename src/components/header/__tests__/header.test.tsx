import { render, screen } from "@testing-library/react";
import Header from "../header";
import React from "react";

describe("Header Component", () => {
    jest.spyOn(console, "warn").mockImplementation(() => {});

    it("renders header with correct title", () => {
        render(<Header />);
        const titleElement = screen.getByAltText(/Demo Next.js Logo/i);
        expect(titleElement).toBeInTheDocument();
    });

    it("snapshots header component", () => {
        const { container } = render(<Header />);
        expect(container).toMatchSnapshot();
    });
});