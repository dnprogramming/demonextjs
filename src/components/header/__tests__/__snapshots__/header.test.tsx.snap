// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Header Component snapshots header component 1`] = `
<div>
  <div
    class="header"
  >
    <a
      href="/"
    >
      <img
        alt="Demo Next.js Logo"
        aria-label="Demo Next.js Logo"
        data-nimg="1"
        decoding="async"
        height="220"
        loading="eager"
        src="/logo.svg"
        style="color: transparent; width: 100%; height: auto;"
        width="620"
      />
    </a>
  </div>
</div>
`;
